browser/js/common/directives/graphs/line-graph.directive.js

```diff
@@ -0,0 +1,115 @@
+app.directive('lineGraph', function(d3Service, $window) {
+    return {
+        restrict: 'EA',
+        scope: {
+            data: "=",
+            cols: "="
+        },
+        link: function(scope, ele, attrs) {
+
+            d3Service.d3().then(function(d3) {
+                let margin = { top: 20, right: 20, bottom: 30, left: 50 },
+                    width = 960 - margin.left - margin.right,
+                    height = 500 - margin.top - margin.bottom,
+                    dataColumns = scope.cols.split(","),
```

Comment:
Why is `scope.cols` coming in as a comma separated string, why not just pass it in as an array? Also keep in mind that if `scope.dataColumns` is simply the result of having split `scope.cols` *right here right now*, then `scope.dataColumns` is not two-way-bound to whatever `scope.cols` references. So if `scope.cols` changes (due to some `ng-click` or whatnot) `scope.dataColumns` will not update—unless this template re-renders, at which point this linking function will run again and this `scope.dataColumns` will get re-assigned.